crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -543,7 +577,53 @@ impl<'a> DependencyProvider for CondaDependencyProvider<'a> {
             dependencies.requirements.push(version_set_id.into());
         }
 
-        for constrains in rec.package_record.constrains.iter() {
+        // If this is a feature-enabled package, add its feature dependencies
+        if let Some(feature_name) = feature {
+            // Find the feature's dependencies
+            if let Some(deps) = record.package_record.optional_depends.get(feature_name) {
+                // Add each dependency for this feature
+                for req in deps {
+                    let version_set_id = match parse_match_spec(
+                        &self.pool,
+                        req,
+                        &mut parse_match_spec_cache,
+                    ) {
+                        Ok(version_set_id) => version_set_id,
+                        Err(e) => {
+                            let reason = self.pool.intern_string(format!(
+                                "the optional dependency '{req}' for feature '{feature_name}' failed to parse: {e}"
+                            ));
+                            return Dependencies::Unknown(reason);
+                        }
+                    };
+                    dependencies.requirements.push(version_set_id.into());
+                }
+
+                // Add a dependency back to the base package with exact version
+                let base_dep = format!(
+                    "{}=={}",
+                    record.package_record.name.as_normalized(),
+                    record.package_record.version
+                );
+
+                let version_set_id = match parse_match_spec(
+                    &self.pool,
+                    base_dep.as_str(),
+                    &mut parse_match_spec_cache,
+                ) {
+                    Ok(version_set_id) => version_set_id,
+                    Err(e) => {
+                        let reason = self.pool.intern_string(format!(
+                            "failed to create base package dependency: {e}",
+                        ));
+                        return Dependencies::Unknown(reason);
+                    }
+                };
```

Comment:
Instead of parsing you could also instantiate a MatchSpec where you set the name, version, build_string and number, and subdir. Or perhaps just the sha256 hash, or all?